ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoBlankLinesInChainedMethodCallsRule.kt

```diff
@@ -15,7 +15,10 @@ public class NoBlankLinesInChainedMethodCallsRule : Rule("no-blank-lines-in-chai
         val isBlankLine = node is PsiWhiteSpace && node.getText().contains("\n\n")
         if (isBlankLine && node.treeParent.elementType == DOT_QUALIFIED_EXPRESSION) {
             emit(node.startOffset + 1, "Needless blank line(s)", true)
-            (node as LeafPsiElement).rawReplaceWithText("\n" + node.getText().split("\n\n")[1])
+
+            if (autoCorrect) {
```

Comment:
I assume we need this check, but I couldn't find a good way to write a test that drives it out. Help would be appreciated!

Reply:
If you mean `if (autoCorrect) {` then you have already done so in test case `single blank line in dot qualified expression returns lint error`. 

When calling `hasLintViolations`, the rule engine is executed with `autocorrect` set to `false` which results in emitting the error without fixing it. When calling 'isFormattedAs`, the rule engine is invoked with `autocorrect` set to `true` resulting in formattting the code.

```
        noBlankLinesInChainedMethodCallsRuleAssertThat(code)
            .hasLintViolations(LintViolation(3, 1, "Needless blank line(s)"))
            .isFormattedAs(
                """
                fun foo(inputText: String) {
                    inputText
                        .toLowerCase()
                }
                """.trimIndent()
            )
            ```

Reply:
Ah right. But, if I remove the `if` statement (and just always remove the blank line), I don't see a failure. I'm assuming we don't want to be messing around with the code when `autocorrect` is `false`. Maybe we need to add a check to `hasLintViolations` to check that the code is unchanged?

Reply:
@paul-dingemans thanks for merging, but still a little concerned there's no coverage for the `if (autocorrect)` check. I'd be happy to look at that in the future if you agree it's an issue?

Reply:
Sorry, being more specific, I mean that it doesn't have "branch" coverage - it does indeed have "line" coverage. If I change the `if` statement to just be `if (true)` for example (or just remove the `if` and keep the `rawReplaceWithText`), the tests will still all pass. There's no test that ensures that we only run reformat when `autoCorrect` is `true`.

Reply:
You don't have to worry about that. There is not a single rule for which this is being tested.